pkg/cloud/services/networking/port.go

```diff
@@ -265,7 +275,27 @@ func (s *Service) DeletePort(eventObject runtime.Object, portID string) error {
 	return nil
 }
 
-func (s *Service) DeletePorts(openStackCluster *infrav1.OpenStackCluster) error {
+// DeleteTrunk deletes the Neutron trunk and port with the given ID.
+func (s *Service) DeleteInstanceTrunkAndPort(eventObject runtime.Object, port infrav1.PortStatus) error {
+	trunkSupported, err := s.isTrunkExtSupported()
```

Comment:
Oh, we don't cache this, this is additional API call per each deleted port. Maybe it's not related to the patch, but pretty ugly.

Reply:
It's ugly. We could maybe cache it in OpenStackClusterStatus, I don't know

Reply:
Previously the code was only doing this once per instance [1]. Could we at least call this in the upper loop?

[1] https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/1788/files#diff-5e7dbfd52305a21a11fb94fab201062ae2c7b851db93c2a8443a1d82d7aa7936L650-L653

Reply:
I've made it so it's called at minimum times in upper loops.